server/src/main/java/org/opensearch/telemetry/tracing/channels/TraceableHttpChannel.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.channels;
+
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.http.HttpChannel;
+import org.opensearch.http.HttpResponse;
+import org.opensearch.telemetry.tracing.Span;
+import org.opensearch.telemetry.tracing.listener.TraceableActionListener;
+
+import java.net.InetSocketAddress;
+import java.util.Objects;
+
+/**
+ * Tracer wrapped {@link HttpChannel}
+ */
+public class TraceableHttpChannel implements HttpChannel {
```

Comment:
@Gaganjuneja one hidden issue related to this feature is that we are going to generate a lot for wrappers when the  `FeatureFlags.isEnabled(TELEMETRY)` is not set. I think we should take the factory method approach and return delegate in such cases.

Reply:
It make sense, should we do for the case where FeatureFlags.isEnabled(TELEMETRY)== true but TelemetrySettings.TRACER_ENABLED_SETTING is false. Later would be little difficult as we need the access of Settings object.

Reply:
The TelemetrySettings.TRACER_ENABLED_SETTING is dynamic, we should not rely on it